### PR TITLE
Fixes various issues in Mouse class and implements more functions of it.

### PR DESCRIPTION
**Pull request by tomsons26**

### CODE REVIEW

File: src/game/client/input/mouse.cpp [hunk 1]
```diff
@@ -715,3 +741,30 @@ void Mouse::Parse_Cursor_Definitions(INI *ini)
         ini->Init_From_INI(&g_theMouse->m_cursorInfo[g_theMouse->Get_Cursor_Index(tok)], _cursor_parsers);
     }
 }
+
+bool Mouse::Is_Click(ICoord2D *coord1, ICoord2D *coord2, unsigned val1, unsigned val2) const
```
Comment (duncanspumpkin) [inline src/game/client/input/mouse.cpp]: Val 1 and 2 are probably time 1 and time 2

File: src/platform/input/win32mouse.cpp [hunk 2]
```diff
@@ -108,7 +108,7 @@ void Win32Mouse::Set_Visibility(bool visibility)
 }
 
 // 0x0073BE20
-bool Win32Mouse::Get_Mouse_Event(MouseIO *io, int8_t unk)
+uint8_t Win32Mouse::Get_Mouse_Event(MouseIO *io, int8_t unk)
 {
```
Comment (duncanspumpkin) [inline src/platform/input/win32mouse.cpp:112]: Could you do a check and see if the false should be changed to 0xff

File: src/game/client/input/mouse.cpp [hunk 3]
```diff
@@ -74,10 +77,29 @@ const char *Mouse::s_cursorNames[] = {
...
+    image_name.Clear();
+    w3d_model_name.Clear();
+    w3d_anim_name.Clear();
+}
+
 /**
  * 0x004029D0
  */
 Mouse::Mouse() :
+    m_cursorInfo(),
```
Comment (OmniBlade) [inline src/game/client/input/mouse.cpp]: No need to have this in the initialiser list, the array will always be default initialised by calling the default constructor for each item in the array.

**Review by OmniBlade — APPROVED** [review]
You can remove the array initialisation from the Mouse constructor, but since it seems to compile as is I suppose it doesn't really matter. Looks good otherwise, subject to the other suggestions being implemented as well.

**Review by OmniBlade — APPROVED** [review]

**Pull request merged**